src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -354,7 +354,7 @@ private async Task StartAsyncCore(CancellationToken cancellationToken)
                 Log.Starting(_logger);
 
                 // Start the connection
-                var connection = await _connectionFactory.ConnectAsync(_protocol.TransferFormat);
+                var connection = await _connectionFactory.ConnectAsync(_protocol.TransferFormat, cancellationToken);
```

Comment:
We probably want to link to the new _state.StopCts.Token like [we just started doing in HandshakeAsync](https://github.com/aspnet/AspNetCore/blob/master/src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs#L1026) so StopAsync can short circuit the start/reconnect attempt faster.

Reply:
Today's commit contains the requested changes.